### PR TITLE
Feat/obs chart

### PR DESCRIPTION
**Pull request by thmslhst**
# What does this PR do?

This PR focuses on improving and extending the charts. All charts are based on [Recharts](https://recharts.org/en-US) and **must** be placed inside a **ChartContainer** to benefit from its actions (isLoading, isEmpty), following a composition pattern.

```
export const AreaChartExample = {
  render: () => (
    <Chart.Container className="h-[300px] w-full p-5 py-2 pr-0">
      <ComposedChart data={sampleData} margin={{ top: 14, bottom: 0, left: 0, right: 0 }}>
        <CartesianGrid horizontal={true} vertical={false} stroke="var(--color-neutral-250)" />
        <XAxis
          dataKey="timestamp"
          type="number"
          scale="time"
          domain={[1704067200000, 1704088800000]}
          tick={{ fontSize: 12, fill: 'var(--color-neutral-350)' }}
          tickLine={{ stroke: 'transparent' }}
          axisLine={{ stroke: 'var(--color-neutral-250)' }}
          tickFormatter={(timestamp) => {
            const date = new Date(timestamp)
            const hours = date.getHours().toString().padStart(2, '0')
            const minutes = date.getMinutes().toString().padStart(2, '0')
            return `${hours}:${minutes}`
          }}
          strokeDasharray="3 3"
        />
        <YAxis
          tick={{ fontSize: 12, fill: 'var(--color-neutral-350)' }}
          tickLine={{ stroke: 'transparent' }}
          axisLine={{ stroke: 'transparent' }}
          orientation="right"
          tickCount={5}
          tickFormatter={(value) => value === 0 ? '' : value}
        />
        <Chart.Tooltip content={<Chart.TooltipContent title="Usage Metrics" />} />
        <Area type="linear" dataKey="value" stackId="httpErrors" stroke="var(--color-brand-500)" fill="var(--color-brand-500)" fillOpacity={0.6} strokeWidth={2} isAnimationActive={false} />
        <Area type="linear" dataKey="secondary" stackId="httpErrors" stroke="var(--color-purple-500)" fill="var(--color-purple-500)" fillOpacity={0.6} strokeWidth={2} isAnimationActive={false} />
      </ComposedChart>
    </Chart.Container>
  ),
}
```

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview e12a2902-8071-4d9f-aa36-7cbfb6252d14 | [feat/obs](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/e12a2902-8071-4d9f-aa36-7cbfb6252d14/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> You are going to thank you later if it contains tests

File: libs/domains/observability/feature/src/lib/service-overview/cpu-chart/cpu-chart.tsx [hunk 1]
```diff
@@ -79,6 +79,20 @@ export function CpuChart({ clusterId, serviceId }: { clusterId: string; serviceI
     return metrics.data.result.map((_: unknown, index: number) => metrics.data.result[index].metric.pod) as string[]
   }, [metrics])
 
+  const renderCpuLimitLabel = (props: any) => {
```
Comment (RemiBonnet) [inline libs/domains/observability/feature/src/lib/service-overview/cpu-chart/cpu-chart.tsx]: We need to find a way to remove this `any`

File: libs/shared/ui/src/lib/components/chart/chart-axis-utils.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+function getLogicalTicks(startTimestamp: number, endTimestamp: number, tickCount = 6): number[] {
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/chart/chart-axis-utils.ts]: Tests are needed for this element and you can call it `chart-util.ts` for now to have something more generic

File: libs/shared/ui/src/lib/components/chart/chart.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+  { timestamp: 1704070800000, time: '01:00', fullTime: '2024-01-01 01:00:00', cpu: 45, memory: 72, disk: 47 },
+  { timestamp: 1704074400000, time: '02:00', fullTime: '2024-01-01 02:00:00', cpu: 78, memory: 65, disk: 52 },
+  { timestamp: 1704078000000, time: '03:00', fullTime: '2024-01-01 03:00:00', cpu: 92, memory: 58, disk: 46 },
+  { timestamp: 1704081600000, time: '04:00', fullTime: '2024-01-01 04:00:00', cpu: 68, memory: 82, disk: 49 },
+  { timestamp: 1704085200000, time: '05:00', fullTime: '2024-01-01 05:00:00', cpu: 34, memory: 76, disk: 51 },
+  { timestamp: 1704088800000, time: '06:00', fullTime: '2024-01-01 06:00:00', cpu: 52, memory: 69, disk: 48 },
+]
+
+// Generate 20 unique gaffotron instance metrics
+const generateGaffotronMetrics = () => {
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/chart/chart.stories.tsx]: `Gaffotron` is a random name of the service you can rename it to `generateMetrics`

File: libs/shared/ui/src/lib/components/chart/chart.test.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
+import { ComposedChart, Line } from 'recharts'
+import { render, screen } from '@qovery/shared/util-tests'
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/chart/chart.test.tsx]: Prefer to use `renderWithProviders` instead of `render`

File: libs/shared/ui/src/lib/components/chart/chart.test.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+      const { container } = render(
+        <Chart.Container className="h-[300px] w-full">
+          <ComposedChart data={sampleData}>
+            <Line dataKey="cpu" />
+          </ComposedChart>
+        </Chart.Container>
+      )
+
+      const chartContainer = container.firstChild as HTMLElement
+      expect(chartContainer).toHaveClass('relative', 'flex', 'h-[300px]', 'justify-center', 'text-xs', 'w-full')
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/chart/chart.test.tsx]: Could you not use class for it ? except if you check the result with opacity. I think you can find a role or aria-placeholder or as a last resort `data-testid` 

And the name of the file should be `chart.spec.ts` like other file

File: libs/shared/utils/src/index.ts [hunk 6]
```diff
@@ -1,2 +1,5 @@
 // http
 export * from './lib/http/interceptors/auth-interceptor/auth-interceptor'
+
+// design-system
+export * from './lib/design-system/colors'
```
Comment (RemiBonnet) [inline libs/shared/utils/src/index.ts]: I don't think this really belongs to the design system, since everything related to the design system should be in the UI library. These colors are related to charts/instances, so I think we could use something more generic

For now, you can add it to the /chart folder as chart-colors.ts and name it CHART_COLORS. We’ll see later how to manage it more cleanly

Also, the /shared library is deprecated, we’d like to remove it and replace it with more specific libraries like util-date, etc.

File: libs/shared/ui/src/lib/components/chart/chart.tsx [hunk 7]
```diff
@@ -62,8 +55,14 @@ const ChartTooltipContent = forwardRef<
     title: string
     formatValue?: (value: string, dataKey: string) => string
     formatLabel?: (dataKey: string) => string
+    maxItems?: number
   }
->(({ active, payload, title, formatValue, formatLabel }, ref) => {
+>(function ChartTooltipContent({ active, payload, title, formatValue, formatLabel, maxItems }, ref) {
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/chart/chart.tsx]: I think you can add a value by default 
```suggestion
>(function ChartTooltipContent({ active, payload, title, formatValue, formatLabel, maxItems = 15 }, ref) {
```

File: libs/domains/observability/feature/src/lib/service-overview/cpu-chart/cpu-chart.tsx (hunk 1 above)
Comment (thmslhst) [reply libs/domains/observability/feature/src/lib/service-overview/cpu-chart/cpu-chart.tsx]: I've updated renderCpuLimitLabel to a more generic renderResourceLimitLabel, member of LocalChart. Added a proper type for the props too. So now we can use it with all our obs charts

File: libs/shared/ui/src/lib/components/chart/chart-axis-utils.ts (hunk 2 above)
Comment (thmslhst) [reply libs/shared/ui/src/lib/components/chart/chart-axis-utils.ts]: Done here: f82e286d8b11c197b99d3791c0325f024091b272

We test for:
1. Default tick count behavior (6 ticks)
2. Custom tick count
3. Even spacing verification
4. Single tick edge case (returns NaN due to division by zero)
5. Zero duration handling

File: libs/shared/ui/src/lib/components/chart/chart.test.tsx (hunk 5 above)
Comment (thmslhst) [reply libs/shared/ui/src/lib/components/chart/chart.test.tsx]: Ok so we now check via: role="region" aria-label="Interactive chart" -- what do you think about it?

File: libs/shared/ui/src/lib/components/chart/chart.stories.tsx (hunk 3 above)
Comment (thmslhst) [reply libs/shared/ui/src/lib/components/chart/chart.stories.tsx]: Done here: 80c476bdb3cc6405df2c7038d889595718e95f80

File: libs/shared/ui/src/lib/components/chart/chart.test.tsx (hunk 4 above)
Comment (thmslhst) [reply libs/shared/ui/src/lib/components/chart/chart.test.tsx]: Done here: cc4fde37375ca745e899bc8ba0166843e494b254

File: libs/shared/ui/src/lib/components/chart/chart.test.tsx (hunk 5 above)
Comment (RemiBonnet) [reply libs/shared/ui/src/lib/components/chart/chart.test.tsx]: Yes it should be good !

File: libs/shared/utils/src/index.ts (hunk 6 above)
Comment (thmslhst) [reply libs/shared/utils/src/index.ts]: Ok! I moved the palette to the /chart folder: 1461128bd0218e23d1abe5913dbeca7a820a4dea
I haven't changed the palette itself, so it's still derived from main.css colors for now...

File: libs/shared/ui/src/lib/components/chart/chart.tsx (hunk 7 above)
Comment (thmslhst) [reply libs/shared/ui/src/lib/components/chart/chart.tsx]: Updated as part of this commit: 19644838b39e87f7561012c81246d4fe544216b1
